1_fetch/src/fetch_wqp_data.R

```diff
@@ -5,22 +5,49 @@
 #' 
 #' @param sitecounts_df data frame containing the site identifiers and total number of
 #' records available for each site. Must contain columns `MonitoringLocationIdentifier`.
-#' @param max_sites_allowed integer indicating the maximum number of sites allowed
-#' in each download group. Defaults to 500.
+#' @param max_sites integer indicating the maximum number of sites allowed in each
+#' download group. Defaults to 500.
+#' @param max_results integer indicating the maximum number of records allowed in
+#' each download group. Defaults to 250,000.
 #' 
 #' @return returns a data frame with columns site id, the total number of records,
 #' (retains the column from `sitecounts_df`), site number, and an additional column 
 #' called `download_grp` which is made up of unique groups that enable use of 
 #' `group_by()` and then `tar_group()` for downloading.
 #' 
-add_download_groups <- function(sitecounts_df, max_sites_allowed = 500) {
+add_download_groups <- function(sitecounts_df, max_sites = 500, max_results = 250000) {
   
+  # Check whether any individual sites have a records count that exceeds `max_results`
+  if(any(sitecounts_df$results_count > max_results)){
+    sites_w_many_records <- sitecounts_df %>%
+      filter(results_count > max_results) %>%
+      pull(MonitoringLocationIdentifier)
+    # Print a message to inform the user that some sites contain a lot of data
+    message(sprintf(paste0("results_count exceeds max_results for the sites ",
+                           "below. If you are not already branching across ",
+                           "characteristic names, consider doing so to set up ",
+                           "a manageable query to WQP.\n\n%s\n"),
```

Comment:
Should we also mention the potential to increase `max_results` ?

Reply:
We could definitely do that, although I do wonder if mentioning the potential to increase `max_results` in this message would encourage a user to then create larger queries, perhaps just so that this message goes away 🤷

Reply:
What if we said something that describes the implication of `results_count` > `max_results`, like:

> "results_count exceeds max_results for the sites below. Assigning data-heavy sites to their own download group to set up a manageable query to WQP. If you are not already branching across characteristic names, consider doing so to further limit query size."

Reply:
Excellent point, I would likely be that kind of person :)

Reply:
I think that message captures everything nicely!

Reply:
Wonderful! I've added that text in [beb377f](https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/58/commits/beb377ff9ab068fe65c319c24e75b1f2db0eaaff) along with some other minor edits to comments.